### ISSUE TITLE
Suggestion to Include `torch.no_grad()` Context During Inference

### ISSUE THREAD

**Issue by Ethan-Chen-plus** [hkchengrex/XMem#151]
Hello, and thank you for this impressive repository!

I noticed that in the provided inference code(colab):https://colab.research.google.com/drive/1RXK5QsUo2-CnOiy5AOSjoZggPVHOPh1m?usp=sharing#scrollTo=mu8fe8V4NFrh, the `torch.cuda.amp.autocast(enabled=True)` context is used for automatic mixed precision, which is excellent for optimizing performance. However, I observed that `torch.no_grad()` is not included here. Typically, in inference scenarios, wrapping the code in a `torch.no_grad()` context can help reduce memory usage by disabling gradient calculations, leading to potentially faster execution and lower memory consumption.

Could you please consider the following modification to enhance memory efficiency?

**Code Reference:**

```python
with torch.cuda.amp.autocast(enabled=True):
    while (cap.isOpened()):
        # Frame processing code
```

**Suggested Modification:**

For improved inference efficiency, please consider adding `torch.no_grad()` as an outer context to disable gradients entirely during this stage. This can be implemented as follows:

```python
with torch.no_grad():
    with torch.cuda.amp.autocast(enabled=True):
        while (cap.isOpened()):
            # Frame processing code
```

If the correct approach is not to use `torch.no_grad`, could you please explain the reasoning? Thank you!

**Comment by hkchengrex** [comment]
Hi, thanks for bringing this up!

We have set `torch.set_grad_enabled(False)` which should do the same thing. Feel free to let me know if it doesn't and needs fixing.
